cmd/term-pager.go

```diff
@@ -100,10 +101,31 @@ func (m model) headerView() string {
 }
 
 func (m model) footerView() string {
+	// Disables printing if the viewport is not ready
+	if m.viewport.Width == 0 {
+		return ""
+	}
+
 	percent := percentStyle.Render(fmt.Sprintf("%d%%", int(m.viewport.ScrollPercent()*100)))
 	info := fmt.Sprintf(" %s", percent)
-	line := strings.Repeat("─", max(0, m.viewport.Width-lipgloss.Width(info)))
-	return lipgloss.JoinHorizontal(lipgloss.Center, line, info)
+	totalLength := m.viewport.Width - lipgloss.Width(info)
+	var finishedCount int
+	var onePart int
+	if totalLength <= 99 {
+		onePart = m.viewport.Width / 5
+		if !math.IsNaN(m.viewport.ScrollPercent()) {
+			finishedCount = onePart * min(5, int(m.viewport.ScrollPercent()*100)/20)
+		}
+	} else {
+		onePart = max(1, totalLength/100)
```

Comment:
I think onePart needs to be float so the percentage calculation will be more correct. Can you check 100% here with different terminal widh in your machine ? it does not work very well for me

Reply:
The precision won't be better since it's all ints. Even if I transform the ints to floats they still need to go back. and it's a 1-100 so decimals won't really matter. 

However, there is a big in it .. the part where I break it down into 5 parts to make it fit smaller terminals better seems to have been done wrong so it's using 1/5 paging for all sizes. 

I will fix that on monday

Reply:
You were right, a float conversion gave me the accuracy needed for both small and big terminals. Thank you.